### PR TITLE
Diagnose associated types inside '@objc' protocols

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: include/swift/AST/DiagnosticsSema.def [hunk 1]
```diff
@@ -1749,6 +1749,10 @@ WARNING(associated_type_override_typealias,none,
         "associated type %0 is redundant with type %0 declared in inherited "
         "%1 %2", (DeclName, DescriptiveDeclKind, Type))
 
+ERROR(associated_type_objc,none,
+      "associated type %0 cannot be declared inside '@objc' protocol %1",
```
Comment (jrose-apple) [inline include/swift/AST/DiagnosticsSema.def]: Can you rephrase this to make it clearer that they're not allowed at all? In particular, mentioning the names makes it sounds like a problem specific to this protocol or this associated type.
